rts/Game/SelectedUnitsAI.cpp

```diff
@@ -492,7 +492,7 @@ void CSelectedUnitsHandlerAI::CreateUnitOrder(std::vector< std::pair<float, int>
 
 		// give weaponless units a long range to make them go to the back
 		const float range = (unit->maxRange < 1.0f)? 2000: unit->maxRange;
-		const float value = ((ud->metal * 60) + ud->energy) / ud->health * range;
+		const float value = ((ud->cost.metal * 60) + ud->cost.energy) / ud->health * range;
```

Comment:
Maybe add `SResourcePack::ToMetalCost() and SResourcePack::ToEnergyCost()` to account for the magic `metal + energy/60` and `(metal + energy/60) * 60` magic formulas here and one item below.

Reply:
I planned to remove conversions like this altogether and just replace it with `ud->power` which defaults to `metal + energy / 60` and which better models what is wanted here. But later on because that's a logic change and so far this PR doesn't have any non-trivial ones.